ui/components/MesheryPerformance/PerformanceCard.js

```diff
@@ -45,43 +48,143 @@ const useStyles= makeStyles((theme) => ({
     alignItems : "center",
   },
   lastRunText : {
-    marginRight : "0.5rem"
+    marginRight : "0.5rem",
+    marginLeft : "0.5rem",
   },
   resultText : {
     color : theme.palette.secondary.lightText,
   },
 }))
 
+const avatarHandler = (function AvatarHandler() {
+const idToAvatarMap = {}
+
+function fetchUserAvatarLink(userId, setterCallbackFunction) {
+  if (!userId) return null
+  dataFetch(`/api/user/profile/${userId}`, {
+    credentials : "include"
+  },
+  function assignAvatarLinkToId(result) {
+    if (result.avatar_url) {
+      idToAvatarMap[userId] = result.avatar_url
+      setterCallbackFunction(result.avatar_url)
+    }
+  },
+  function handleProfileFetchError(error) {
+    console.error("failed to fetch user profile with ID", userId, error)
+  }
```

Comment:
I wish to know what will be the expected behavior if this function will be called, as If any error occurs, instead of console.error, shall we display the error with Toaster or something similar for better UX? 
Am I missing something here?